src/app/clusters/commissioner-control-server/commissioner-control-server.cpp

```diff
@@ -0,0 +1,251 @@
+                                                                            .productId    = productId,
+                                                                            .clientNodeId = sourceNodeId,
+                                                                            .fabricIndex  = fabricIndex,
+                                                                            .label        = label };
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->HandleCommissioningApprovalRequest(request);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "emberAfCommissionerControlClusterRequestCommissioningApprovalCallback error: %s", err.AsString());
+        status = Status::Failure;
+    }
+
+    commandObj->AddStatus(commandPath, status);
+    return true;
+}
+
+bool emberAfCommissionerControlClusterCommissionNodeCallback(
+    app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+    const Clusters::CommissionerControl::Commands::CommissionNode::DecodableType & commandData)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    Status status  = Status::Success;
+
+    ChipLogProgress(Zcl, "Received command to commission node");
+
+    auto sourceNodeId = getNodeId(commandObj);
+    auto requestId    = commandData.requestId;
+
+    auto info       = std::make_unique<Clusters::CommissionerControl::CommissionNodeInfo>();
+    info->ipAddress = commandData.ipAddress;
+    info->port      = commandData.port;
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->ValidateCommissionNodeCommand(sourceNodeId, requestId, info->params);
+    VerifyOrExit(err == CHIP_NO_ERROR, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Add the response for the commissioning window
+    AddReverseOpenCommissioningWindowResponse(commandObj, commandPath, info->params);
+
+    // Schedule the deferred reverse commission node task
+    DeviceLayer::PlatformMgr().ScheduleWork(RunDeferredCommissionNode, reinterpret_cast<intptr_t>(info.release()));
+
+    return true;
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "emberAfCommissionerControlClusterCommissionNodeCallback error: %s", err.AsString());
+        status = Status::Failure;
+    }
+
+    commandObj->AddStatus(commandPath, status);
```

Comment:
You have already added a response in `AddReverseOpenCommissioningWindowResponse` in some instance here.

I think you only want to call AddStatus in the failure case (if statement above)

Reply:
Oh...

I see what happened you have return true before the exit here... Um I thought the exit para-dime coding practice was that all code needed to end up in the exit even the success case.

Right now only error code ends up in the exit section. Right?